google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableTracerStreamingCallable.java

```diff
@@ -60,7 +67,15 @@ public void call(
       BigtableTracerResponseObserver<ResponseT> innerObserver =
           new BigtableTracerResponseObserver<>(
               responseObserver, (BigtableTracer) context.getTracer(), responseMetadata);
-      innerCallable.call(request, innerObserver, responseMetadata.addHandlers(context));
+      if (batching) {
+        innerCallable.call(request, innerObserver, responseMetadata.addHandlers(context));
+      } else {
+        innerCallable.call(
+            request,
+            innerObserver,
+            Util.addHandlerToCallContext(
+                context, responseMetadata, (BigtableTracer) context.getTracer()));
```

Comment:
The difference here is too subtle.
1. extract the difference out:
if (batching) {
  ctx = Util.addHandlerToCallContext(
                context, responseMetadata, (BigtableTracer) context.getTracer()))
} else {
  ctx = responseMetadata.addHandlers(context)
}

innerCallable.call(request, observer, ctx).


2. Add comments to the if statement as to why it needs different handling